File: 0525-contiguous-array/0525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        Map<Integer,Integer> hm=new HashMap<>();
        hm.put(0,-1);
        int sum=0;
        int maxLength=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                sum-=1;
            }
            else{
                sum+=1;
            }
            if(!hm.containsKey(sum)){
                hm.put(sum,i);
            }
            else{
                maxLength=Math.max(i-hm.get(sum),maxLength);
            }
        }
        return maxLength==Integer.MIN_VALUE?0:maxLength;
    }
}

File: 0525-contiguous-array/0525-contiguous-array.py

class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        my_dict={}
        sum=0;
        my_dict[sum]=-1
        res=0;
        
        for i,n in enumerate(nums):
            if n==0:
                sum-=1
            else:
                sum+=1
            if sum in my_dict:
                res=max(i-my_dict[sum],res)
            else:
                my_dict[sum]=i
        return res;